diff --git a/src/Plugins/VerilatorPlugin/VerilatorIntegrationLibrary/src/renode_bus.cpp b/src/Plugins/VerilatorPlugin/VerilatorIntegrationLibrary/src/renode_bus.cpp
index 24eb4172..0daa8add 100644
--- a/src/Plugins/VerilatorPlugin/VerilatorIntegrationLibrary/src/renode_bus.cpp
+++ b/src/Plugins/VerilatorPlugin/VerilatorIntegrationLibrary/src/renode_bus.cpp
@@ -173,6 +173,7 @@ void RenodeAgent::handleInterrupts(void)
     }
 }
 
+#include <unistd.h>
 void RenodeAgent::simulate(int receiverPort, int senderPort, const char* address)
 {
     renodeAgent = this;
@@ -182,7 +183,10 @@ void RenodeAgent::simulate(int receiverPort, int senderPort, const char* address
     Protocol* result;
     reset();
 
+    int iter = 0;
     while(channel->getIsConnected()) {
+        if(++iter > 1000) sleep(600);
+
         result = receive();
         handleRequest(result);
         delete result;
