diff --git a/src/Plugins/CoSimulationPlugin/IntegrationLibrary/src/renode_bus.cpp b/src/Plugins/CoSimulationPlugin/IntegrationLibrary/src/renode_bus.cpp
--- a/src/Plugins/CoSimulationPlugin/IntegrationLibrary/src/renode_bus.cpp
+++ b/src/Plugins/CoSimulationPlugin/IntegrationLibrary/src/renode_bus.cpp
@@ -198,8 +198,11 @@
+#include <unistd.h>
 void RenodeAgent::simulate()
 {
     Protocol* result;
     reset();
 
+    int iter = 0;
     while(communicationChannel->isConnected()) {
+        if(++iter > 1000) sleep(600);
         result = receive();
         handleRequest(result);
